src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -1079,15 +1079,28 @@ extern "C" int32_t SystemNative_CopyFile(intptr_t sourceFd, intptr_t destination
 
     // Now that the data from the file has been copied, copy over metadata
     // from the source file.  First copy the file times.
+    // If futimes nor futimes are available on this platform, file times will
+    // not be copied over.
     while (CheckInterrupted(ret = fstat_(inFd, &sourceStat)));
     if (ret == 0)
     {
+#if HAVE_FUTIMES
         struct timeval origTimes[2];
-        origTimes[0].tv_sec = sourceStat.st_atime;
         origTimes[0].tv_usec = 0;
-        origTimes[1].tv_sec = sourceStat.st_mtime;
         origTimes[1].tv_usec = 0;
+        origTimes[0].tv_sec = sourceStat.st_atime;
+        origTimes[1].tv_sec = sourceStat.st_mtime;
         while (CheckInterrupted(ret = futimes(outFd, origTimes)));
+#elif HAVE_FUTIMENS
+        // futimes is not a POSIX function, and not available on Android,
+        // but futimens is
+        struct timespec origTimes[2];
+        origTimes[0].tv_usec = 0;
+        origTimes[1].tv_usec = 0;
```

Comment:
Yes, that's correct. I kept the original order - that makes the diff slightly smaller in this case. 
Also updated the configure script: it will no longer fail if futimes nor futimens are available. This reflects the file copy operation just skipping setting the file time alltogether.